From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 8 Oct 2017 22:54:00 +0100
Subject: [PATCH] cfe: Use correct Solus multilib paths

The correct native arch libdir on Solus is /usr/lib64, whereas the
multilib 32bit (i686) arch directory is /usr/lib32. The main linkers
can then be found inside these directories.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 clang/lib/Driver/ToolChains/Linux.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/clang/lib/Driver/ToolChains/Linux.cpp b/clang/lib/Driver/ToolChains/Linux.cpp
index 35bf39069605..3674789ab88d 100644
--- a/clang/lib/Driver/ToolChains/Linux.cpp
+++ b/clang/lib/Driver/ToolChains/Linux.cpp
@@ -594,7 +594,7 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     Loader = "ld64.so.1";
     break;
   case llvm::Triple::x86:
-    LibDir = "lib";
+    LibDir = "lib32";
     Loader = "ld-linux.so.2";
     break;
   case llvm::Triple::x86_64: {
@@ -617,7 +617,7 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
       (Triple.getVendor() == llvm::Triple::UnknownVendor ||
        Triple.getVendor() == llvm::Triple::PC))
     return "/usr/" + Triple.str() + "/lib/" + Loader;
-  return "/" + LibDir + "/" + Loader;
+  return "/usr/" + LibDir + "/" + Loader;
 }
 
 void Linux::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
